src/memory/query_memory_control.hpp

```diff
@@ -78,9 +78,20 @@ class QueriesMemoryControl {
   bool IsThreadTracked();
 
  private:
+  struct TransactionId {
+    uint64_t id;
+    uint64_t cnt;
```

Comment:
Multiple threads can bind to same transaction at same time, so I would set that to atomic

Reply:
I think a thread will only access TransactionId with its own thread id -> it can't be accessed by more than one thread since thread id is unique so all accesses to one object should be sequential.

Reply:
Yes you are right, I thought this was different counter, for transaction id to thread id, but it was thread to transaction mapping which is fine